### PR TITLE
feat: support api prefix for graphql playground

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: support api prefix for graphql playground

#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: graphql-server/go-server/main.go [hunk 1]
```diff
@@ -70,7 +72,15 @@ func main() {
 	})
 
 	if *enablePlayground {
-		http.Handle("/", playground.Handler("Arcadia-Graphql-Server", "/bff"))
+		endpoint := "/bff"
+		if *ingressPrefix != "" {
+			prefix := strings.TrimPrefix(*ingressPrefix, "/")
+			prefix = strings.TrimSuffix(prefix, "/")
```
Comment (bjwswang) [inline graphql-server/go-server/main.go:78]: prefix := strings.TrimSuffix(strings.TrimPrefix(*ingressPrefix, "/"), "/")

File: graphql-server/go-server/main.go [hunk 2]
```diff
@@ -70,7 +72,15 @@ func main() {
...
+			if prefix != "" {
```
Comment (bjwswang) [inline graphql-server/go-server/main.go:79]: strings.TrimSuffix(strings.TrimPrefix(*ingressPrefix, "/"), "/")

File: graphql-server/go-server/main.go [hunk 3]
```diff
@@ -70,7 +72,15 @@ func main() {
...
+				endpoint = fmt.Sprintf("/%s%s", prefix, endpoint)
```
Comment (bjwswang) [inline graphql-server/go-server/main.go:80]: endpoint := fmt.Sprintf("%s/bff", strings.TrimSuffix(strings.TrimPrefix(*ingressPrefix, "/"), "/")

@0xff-dev will this work?

File: graphql-server/go-server/main.go [hunk 4]
```diff
@@ -43,6 +44,7 @@ var (
 
 	enablePlayground = flag.Bool("enable-playground", false, "enable the graphql playground")
 	enableOIDC       = flag.Bool("enable-oidc", false, "enable oidc authorization")
+	ingressPrefix    = flag.String("ingress-prefix", "", "this parameter should also be configured when the service is forwarded via ingress and a path prefix is configured to avoid not finding the service, such as /apis")
```
Comment (bjwswang) [inline graphql-server/go-server/main.go:47]: `ingressPrefix` -> `playgroundEndpointPrefix`

File: graphql-server/go-server/main.go (hunk 3 above)
Comment (bjwswang) [reply graphql-server/go-server/main.go:80]: or just pass `--playgroundEndpoint` ???

Comment (0xff-dev) [reply graphql-server/go-server/main.go:80]: https://github.com/99designs/gqlgen/blob/132ec1ce579e9ce3dc772afebf9703c1403d588e/graphql/playground/playground.go#L53

https://github.com/99designs/gqlgen/blob/132ec1ce579e9ce3dc772afebf9703c1403d588e/graphql/playground/playground.go#L118

According to these two functions, endpoint must start with a /. If playgroundEndpointPrefix is written with only a /, the path to endpoint will be incorrect as written above.

```go
prefix := strings.TrimSuffix(strings.TrimPrefix(*ingressPrefix, "/"), "/")
if prefix != "" {
    endpoint = fmt.Sprintf("/%s%s", prefix, endpoint)
}
```

Comment (bjwswang) [reply graphql-server/go-server/main.go:80]: we need a common name for this endpoint definition. How about use `--playgroundEndpoint` with `/bff` by default

**Review by bjwswang — APPROVED** [review]

**Pull request merged**